File: Lab_daalgavar/lab 9/lab9.java
import java.util.Stack;

class Lab9 {
    static class MyQueue {
        private Stack<Integer> stackIn; 
        private Stack<Integer> stackOut; 

        public MyQueue() {
            stackIn = new Stack<>();
            stackOut = new Stack<>();
        }

        public void push(int x) {
            stackIn.push(x);
        }

        public int pop() {
            if (stackOut.isEmpty()) {
                while (!stackIn.isEmpty()) {
                    stackOut.push(stackIn.pop());
                }
            }
            return stackOut.pop();
        }

        public int peek() {
            if (stackOut.isEmpty()) {
                while (!stackIn.isEmpty()) {
                    stackOut.push(stackIn.pop());
                }
            }
            return stackOut.peek();
        }

        public boolean empty() {
            return stackIn.isEmpty() && stackOut.isEmpty();
        }
    }

    public static void main(String[] args) {
        MyQueue myQueue = new MyQueue();

        myQueue.push(1);  
        myQueue.push(2);  

        System.out.println("Peek: " + myQueue.peek()); 
        System.out.println("Pop: " + myQueue.pop());    
        System.out.println("Is empty: " + myQueue.empty()); 
    }
}

File: Lab_daalgavar/lab 9/lab_9.py
class MyQueue:
    def __init__(self):
        self.stack_in = []  
        self.stack_out = []  

    def push(self, x):
        self.stack_in.append(x)  

    def pop(self):
        if not self.stack_out:  
            while self.stack_in:  
                self.stack_out.append(self.stack_in.pop())
        return self.stack_out.pop()  

    def peek(self):
        if not self.stack_out: 
            while self.stack_in: 
                self.stack_out.append(self.stack_in.pop())
        return self.stack_out[-1]  

    def empty(self):
        return not self.stack_in and not self.stack_out  

myQueue = MyQueue()

myQueue.push(1)  
myQueue.push(2)  

print("Peek:", myQueue.peek())  
print("Pop:", myQueue.pop())   
print("Is empty:", myQueue.empty())  